gapic-generator-cloud/templates/cloud/helpers/cloud_helper.rb

```diff
@@ -50,47 +88,145 @@ def client_lro? _service
+      service.parent.name.sub ".proto", "_pb.rb"
+    else
+      ruby_file_path(service).sub ".rb", "_pb.rb"
+    end
+  end
+
+  def service_proto_file_name service
+    service_proto_file_path(service).split("/").last
+  end
+
+  def service_proto_require service
+    service_proto_file_path(service).sub ".rb", ""
+  end
+
+
+  def services_proto_name_full service
+    ruby_namespace service.address
+  end
+
+  def services_proto_file_path service
+    if service.parent
+      service.parent.name.sub ".proto", "_services_pb.rb"
+    else
+      ruby_file_path(service).sub ".rb", "_services_pb.rb"
+    end
+  end
+
+  def services_proto_file_name service
+    services_proto_file_path(service).split("/").last
+  end
+
+  def services_proto_require service
+    services_proto_file_path(service).sub ".rb", ""
+  end
+
+
+  def services_stub_name_full service
+    "#{services_proto_name_full service}::Stub"
+  end
+
+
+  def client_file_path service
+    ruby_file_path(service).sub ".rb", "_client.rb"
+  end
+
+  def client_file_name service
+    client_file_path(service).split("/").last
+  end
+
+  def client_require service
+    client_file_path(service).sub ".rb", ""
+  end
+
+
+  def client_name _service
+    "Client"
   end
 
   def client_name_full service
-    "#{ruby_namespace service.address}Client"
+    "#{service_proto_name_full service}::#{client_name service}"
+  end
+
+  def client_test_file_path service
+    client_file_path(service).sub ".rb", "_test.rb"
   end
 
-  def client_require _service
-    "google/cloud/speech/v1/cloud_speech_pb"
+  def credentials_name _service
+    "Credentials"
   end
 
-  def credentials_require _service
-    "google/cloud/speech/v1/credentials"
+  def credentials_name_full service
+    credentials_namespace = service.address.dup
+    credentials_namespace.pop
+    credentials_namespace.push credentials_name(service)
+    ruby_namespace credentials_namespace
   end
 
-  def client_stub _service
-    "speech_stub"
+  def credentials_require service
+    credentials_namespace = service.address.dup
+    credentials_namespace.pop
+    credentials_namespace.push credentials_name(service)
+    credentials_namespace.map(&:underscore).join "/"
   end
 
-  def client_address _service
-    "speech.googleapis.com"
+  def credentials_file_path service
+    credentials_require(service) + ".rb"
   end
 
-  def client_port _service
-    443
+  def helpers_require service
+    helpers_namespace = service.address.dup
+    helpers_namespace.pop
+    helpers_namespace.push "helpers"
+    helpers_namespace.map(&:underscore).join "/"
   end
 
-  def client_scopes _service
-    ["https://www.googleapis.com/auth/cloud-platform"]
+  def helpers_file_path service
+    helpers_require(service) + ".rb"
   end
 
-  def service_proto_require _service
-    "google/cloud/speech/v1/cloud_speech_services_pb"
+  def client_address service
+    address = service.host
+    address ||= service.parent.parent.protoc_options[:default_host]
```

Comment:
For options that map to a proto annotation option, let's not expose a protoc/configuration interface since we want to be strict about where that information is coming from. If we get user need in the future we can reevaluate then.

Reply:
Do we want to look at the protoc options for anything other than generator type and configuration path then? Should we remove the check against `protoc_options` from everywhere?

Reply:
I agree that we should be strict about where information is coming from, it will make maintenance much easier. I would be in favor of removing the check against protoc_options from everywhere.

Reply:
Also in favor of removing protoc_options check from everywhere.

Reply:
Moving discussion to #71.